libnetwork/sandbox_options.go

```diff
@@ -171,3 +171,13 @@ func OptionLoadBalancer(nid string) SandboxOption {
 		sb.oslTypes = append(sb.oslTypes, osl.SandboxTypeLoadBalancer)
 	}
 }
+
+// OptionCtrTaskCreated returns an option setter that says there will be a call
+// to libnetwork's Sandbox.CtrTaskCreated after container task creation, before
+// it's started. Setting this option when the container has already been started
+// (for an option-refresh) is harmless.
+func OptionCtrTaskCreated() SandboxOption {
```

Comment:
I'd be happy to change the name.

I guess I'm still trying to understand the rules for changing interfaces like this. I thought I'd understood that we treat packages as public interfaces that anybody could be using - so we can't be sure nothing else is "doing a buildkit", using libnetwork the way it does? That's why I figured calling the new phase-2 method would have to be opt-in. (Having free-reign over this sort of change would make life a lot easier though!)

Your third option sounds best by-far ... but still, if anything else is using `WithLibnetwork()` it'd be a breaking change. If that's not an issue, let's do it.

Reply:
The third option would not be a breaking change (even if we ignore the fact that neither `func withLibnetwork` nor its sole caller `func (*Daemon) createSpec` are exported) because the `libnetwork-setkey` reexec would continue to behave exactly the same as it currently does! The only functional difference between the existing behaviour and the "new" behaviour is at which point in the task's lifecycle `func (*Sandbox) SetKey` is called. Anything "doing a buildkit" would continue to invoke it via `libnetwork-setkey` from an OCI hook. We opt into the new behaviour by instead calling `SetKey` programmatically after any ipv6-related sysctls have been applied to the network namespace.

Reply:
(`withLibnetwork` used to be an exported function, but it was un-exported as part of #43980. Regardless, we consider the daemon tree to be de-facto internal.)

Reply:
Oh, yes - sure ... I was looking at the diff you linked to, it deletes `WithLibnetwork()` but it's been un-exported since. Sounds good then.

Reply:
I've rebased my PR so you can cherry-pick https://github.com/corhere/moby/commit/bd99fb03193b607df491aebda40910cc2f0f2514 cleanly into this one.

Reply:
Thank you!